plutus-core/plutus-ir/src/PlutusIR/Transform/RewriteRules/Common.hs

```diff
@@ -0,0 +1,89 @@
+{-# LANGUAGE GADTs           #-}
+{-# LANGUAGE PatternSynonyms #-}
+{-# LANGUAGE TypeOperators   #-}
+module PlutusIR.Transform.RewriteRules.Common
+    ( seqA
+    , seqP
+    , mkFreshTermLet -- from MkPlc
+    , pattern A
+    , pattern B
+    , pattern I
+    ) where
+
+import PlutusCore.Builtin
+import PlutusCore.Quote
+import PlutusIR
+import PlutusIR.Analysis.Builtins
+import PlutusIR.Analysis.VarInfo
+import PlutusIR.MkPir
+import PlutusIR.Purity
+
+{- | A wrapper that can be more easily turned into an infix operator.
+
+e.g. `infixr 5 (***) = seqA binfo vInfo`
+-}
+seqA :: (MonadQuote m, Monoid a, ToBuiltinMeaning uni fun)
+     => BuiltinsInfo uni fun
+     -> VarsInfo tyname Name uni a
+     -> (Type tyname uni a, Term tyname Name uni fun a)
+     ->  m (Term tyname Name uni fun a)
+     -> m (Term tyname Name uni fun a)
+seqA binfo vinfo (a,aT) y = seqOpt binfo vinfo a aT <*> y
+
+
+{- | Another "infix" wrapper where second operand is a Haskell pure value.
+
+e.g. `infixr 5 (***) = seqP binfo vInfo`
+-}
+seqP :: (MonadQuote m, Monoid a, ToBuiltinMeaning uni fun)
+     => BuiltinsInfo uni fun
+     -> VarsInfo tyname Name uni a
+     -> (Type tyname uni a, Term tyname Name uni fun a)
+     -> Term tyname Name uni fun a
+     -> m (Term tyname Name uni fun a)
+seqP binfo vinfo p y = seqA binfo vinfo p (pure y)
+
+-- | An optimized version to omit call to `seq` if left operand `isPure`.
```

Comment:
version of what?

Reply:
FWIW I think this function is pointless. Dead code elimination will remove any pure, unused let-bindings, so you might as well just produce the un-optimized code.